Fix ESLint errors on „npm run lint-node“

```
/core/command/approve.js
  12:44  error  Expected error to be handled  handle-callback-err
  14:30  error  Expected error to be handled  handle-callback-err

/core/command/report.js
  105:32  error  Expected '!==' and instead saw '!='  eqeqeq

/core/util/compare.js
  59:3   error  Return statement should not contain assignment  no-return-assign
  59:10  error  'tests' is not defined                          no-undef

/core/util/runCasper.js
  1:5  error  'path' is assigned a value but never used   no-unused-vars
  3:5  error  'isWin' is assigned a value but never used  no-unused-vars

/capture/genBitmaps.js
  106:24  error  Unnecessary escape character: \-                               no-useless-escape
  256:55  error  Unnecessary escape character: \-                               no-useless-escape
  359:3   error  Split initialized 'var' declarations into multiple statements  one-var
  372:25  error  Unnecessary escape character: \/                               no-useless-escape

✖ 11 problems (11 errors, 0 warnings)
```

<img width="2026" alt="screen shot 2017-03-06 at 16 43 01" src="https://cloud.githubusercontent.com/assets/252113/23617278/e1464f98-028c-11e7-90da-e689af875110.png">

--- a/capture/casper_scripts/onBefore.js
+++ b/capture/casper_scripts/onBefore.js
@@ -1,6 +1,6 @@
 module.exports = function (casper, scenario, vp) {
-  //This script runs before your app loads. Edit here to log-in, load cookies or set other states required for your test.
-  console.log('onBefore.js has run for '+ vp.name + '.');
+  // This script runs before your app loads. Edit here to log-in, load cookies or set other states required for your test.
+  console.log('onBefore.js has run for ' + vp.name + '.');
 };
 
 // // EXAMPLE: LOGIN BEFORE RUNNING TESTS

--- a/capture/casper_scripts/onReady.js
+++ b/capture/casper_scripts/onReady.js
@@ -1,6 +1,6 @@
 module.exports = function (casper, scenario, vp) {
   casper.evaluate(function () {
-    //Your web-app is now loaded. Edit here to simulate user interacions or other state changes.
+    // Your web-app is now loaded. Edit here to simulate user interacions or other state changes.
   });
   console.log('onReady.js has run for: ', vp.name);
 };

--- a/capture/genBitmaps.js
+++ b/capture/genBitmaps.js
@@ -32,7 +32,7 @@ if (!config.paths) {
   config.paths = {};
 }
 
-var outputFormat = "." + (config.outputFormat && config.outputFormat.match(/jpg|jpeg/) || 'png');
+var outputFormat = '.' + (config.outputFormat && config.outputFormat.match(/jpg|jpeg/) || 'png');
 var bitmapsReferencePath = config.paths.bitmaps_reference || 'bitmaps_reference';
 var bitmapsTestPath = config.paths.bitmaps_test || 'bitmaps_test';
 var casperScriptsPath = config.paths.casper_scripts || null;
@@ -94,7 +94,7 @@ function capturePageSelectors (scenarios, viewports, bitmapsReferencePath, bitma
   });// end casper.each scenario
 }
 
-function getFilename(scenarioIndex, scenarioLabel, selectorIndex, selectorLabel, viewportIndex, viewportLabel) {
+function getFilename (scenarioIndex, scenarioLabel, selectorIndex, selectorLabel, viewportIndex, viewportLabel) {
   var fileName = fileNameTemplate
     .replace(/\{configId\}/, configId)
     .replace(/\{scenarioIndex\}/, scenarioIndex)
@@ -103,7 +103,7 @@ function getFilename(scenarioIndex, scenarioLabel, selectorIndex, selectorLabel,
     .replace(/\{selectorLabel\}/, selectorLabel)
     .replace(/\{viewportIndex\}/, viewportIndex)
     .replace(/\{viewportLabel\}/, makeSafe(viewportLabel))
-    .replace(/[^a-z0-9_\-]/gi, ''); // remove anything that's not a letter or a number or dash or underscore.
+    .replace(/[^a-z0-9_-]/gi, ''); // remove anything that's not a letter or a number or dash or underscore.
 
   var extRegExp = new RegExp(outputFormat + '$', 'i');
   if (!extRegExp.test(fileName)) {
@@ -116,7 +116,7 @@ function getFilename(scenarioIndex, scenarioLabel, selectorIndex, selectorLabel,
 function processScenario (casper, scenario, scenarioOrVariantLabel, scenarioLabel, viewports, bitmapsReferencePath, bitmapsTestPath, screenshotDateTime) {
   var scriptTimeout = 20000;
 
-  if(casper.cli.options.user && casper.cli.options.password) {
+  if (casper.cli.options.user && casper.cli.options.password) {
     console.log('Auth User via CLI: ' + casper.cli.options.user);
     casper.setHttpAuth(casper.cli.options.user, casper.cli.options.password);
   }
@@ -203,7 +203,7 @@ function processScenario (casper, scenario, scenarioOrVariantLabel, scenarioLabe
           casper.evaluate(function (o) {
             Array.prototype.forEach.call(document.querySelectorAll(o), function (s, j) {
               s.style.display = 'none';
-              s.classList.add('__86d')
+              s.classList.add('__86d');
             });
           }, o);
         });
@@ -216,13 +216,13 @@ function processScenario (casper, scenario, scenarioOrVariantLabel, scenarioLabe
       }
 
       if (scenario.selectorExpansion) {
-        scenario.selectorsExpanded = scenario.selectors.reduce(function(acc, selector) {
+        scenario.selectorsExpanded = scenario.selectors.reduce(function (acc, selector) {
           if (selector === DOCUMENT_SELECTOR) {
-            return acc.concat([DOCUMENT_SELECTOR])
+            return acc.concat([DOCUMENT_SELECTOR]);
           }
 
-          var expandedSelector = casper.evaluate(function(selector) {
-            return [].slice.call(document.querySelectorAll(selector)).map(function(element, expandedIndex) {
+          var expandedSelector = casper.evaluate(function (selector) {
+            return [].slice.call(document.querySelectorAll(selector)).map(function (element, expandedIndex) {
               var indexPartial = '__n' + expandedIndex;
 
               if (element.classList.contains('__86d')) {
@@ -245,7 +245,7 @@ function processScenario (casper, scenario, scenarioOrVariantLabel, scenarioLabe
 
           // concat arrays of fully-qualified classnames
           return acc.concat(expandedSelector);
-        }, []).filter(function(selector) {
+        }, []).filter(function (selector) {
           return selector !== '';
         });
       } else {
@@ -253,7 +253,7 @@ function processScenario (casper, scenario, scenarioOrVariantLabel, scenarioLabe
       }
 
       scenario.selectorsExpanded.forEach(function (o, i, a) {
-        var cleanedSelectorName = o.replace(/[^a-z0-9_\-]/gi, ''); // remove anything that's not a letter or a number
+        var cleanedSelectorName = o.replace(/[^a-z0-9_-]/gi, ''); // remove anything that's not a letter or a number
 
         var fileName = getFilename(scenario.sIndex, scenarioOrVariantLabel, i, cleanedSelectorName, viewportIndex, vp.name);
 
@@ -280,14 +280,12 @@ function processScenario (casper, scenario, scenarioOrVariantLabel, scenarioLabe
   });// end casper.each viewports
 }
 
-function getMisMatchThreshHold(scenario) {
-  if (typeof scenario.misMatchThreshold != 'undefined')
-    return scenario.misMatchThreshold;
+function getMisMatchThreshHold (scenario) {
+  if (typeof scenario.misMatchThreshold !== 'undefined') { return scenario.misMatchThreshold; }
 
-  if (typeof config.misMatchThreshold != 'undefined')
-    return config.misMatchThreshold;
+  if (typeof config.misMatchThreshold !== 'undefined') { return config.misMatchThreshold; }
 
-  return config.defaultMisMatchThreshold
+  return config.defaultMisMatchThreshold;
 }
 
 function captureScreenshot (casper, filePath, selector) {
@@ -358,7 +356,10 @@ function glueStringsWithSlash (stringA, stringB) {
 }
 
 function genHash (str) {
-  var hash = 0, i, chr, len;
+  var hash = 0;
+  var i;
+  var chr;
+  var len;
   if (!str) return hash;
   str = str.toString();
   for (i = 0, len = str.length; i < len; i++) {
@@ -367,9 +368,9 @@ function genHash (str) {
     hash |= 0; // Convert to 32bit integer
   }
   // return a string and replace a negative sign with a zero
-  return hash.toString().replace(/^-/,0);
+  return hash.toString().replace(/^-/, 0);
 }
 
 function makeSafe (str) {
-  return str.replace(/[ \/]/g, '_');
+  return str.replace(/[ /]/g, '_');
 }

--- a/cli/index.js
+++ b/cli/index.js
@@ -46,8 +46,8 @@ if (!commandName) {
     process.exit(exitCode);
   });
 
-  process.on('uncaughtException', function(err) {
-    console.log("Uncaught exception:", err.message, err.stack);
+  process.on('uncaughtException', function (err) {
+    console.log('Uncaught exception:', err.message, err.stack);
     throw err;
   });
 }

--- a/cli/usage.js
+++ b/cli/usage.js
@@ -5,7 +5,7 @@ var commandsDescription = {
   reference: 'Create reference screenshots of your web content at multiple sceen sizes.',
   test: 'Create test screenshots of your web content and compare against the set you created using `backstop reference`.',
   genConfig: 'Generate a configuration file boilerplate in your current directory. PLEASE NOTE: this will force overwrite any existing config.',
-  openReport: 'View your last test screenshots in your browser.',
+  openReport: 'View your last test screenshots in your browser.'
 };
 
 var optionsDescription = {
@@ -22,7 +22,7 @@ function makeDescription (descriptions) {
     .join('\n');
 }
 
-function spacesBetweenCommandAndDescription(commandName) {
+function spacesBetweenCommandAndDescription (commandName) {
   return makeSpaces(2 + leftPaddingOfDescription - commandName.length);
 }
 

--- a/core/command/approve.js
+++ b/core/command/approve.js
@@ -2,16 +2,22 @@ var fs = require('../util/fs');
 var path = require('path');
 var map = require('p-map');
 
-var FAILED_DIFF_RE = /^failed_diff/
+var FAILED_DIFF_RE = /^failed_diff/;
 
 // This task will copy ALL test bitmap files (from the most recent test directory) to the reference directory overwritting any exisiting files.
 module.exports = {
   execute: function (config) {
     // TODO:  IF Exists config.bitmaps_test  &&  list.length > 0n  (otherwise throw)
     console.log('Copying from ' + config.bitmaps_test + ' to ' + config.bitmaps_reference + '.');
-    return fs.readdir(config.bitmaps_test,(err, list) => {
-      var src = path.join(config.bitmaps_test, list[list.length-1]);
+    return fs.readdir(config.bitmaps_test, (err, list) => {
+      if (err) {
+        console.log(err.stack);
+      }
+      var src = path.join(config.bitmaps_test, list[list.length - 1]);
       return fs.readdir(src, (err, files) => {
+        if (err) {
+          console.log(err.stack);
+        }
         console.log('The following files will be promoted to reference...');
         return map(files, (file) => {
           if (FAILED_DIFF_RE.test(file)) {
@@ -19,7 +25,7 @@ module.exports = {
           }
           console.log('> ', file);
           return fs.copy(path.join(src, file), path.join(config.bitmaps_reference, file));
-        })
+        });
       });
     });
   }

--- a/core/command/openReport.js
+++ b/core/command/openReport.js
@@ -4,7 +4,6 @@ var path = require('path');
 
 module.exports = {
   execute: function (config) {
-
     function toAbsolute (p) {
       if (path.isAbsolute(p)) {
         return p;

--- a/core/command/report.js
+++ b/core/command/report.js
@@ -21,7 +21,7 @@ function writeReport (config, reporter) {
 
 function writeBrowserReport (config, reporter) {
   function toAbsolute (p) {
-    return (path.isAbsolute(p))? p : path.join(config.projectPath, p);
+    return (path.isAbsolute(p)) ? p : path.join(config.projectPath, p);
   }
   logger.log('Writing browser report');
   return fs.copy(config.comparePath, toAbsolute(config.html_report)).then(function () {
@@ -98,12 +98,12 @@ module.exports = {
       var failed = report.failed();
       logger.log('Test completed...');
       logger.log(chalk.green(report.passed() + ' Passed'));
-      logger.log(chalk[(failed ? 'red' : 'green')]( + failed + ' Failed'));
+      logger.log(chalk[(failed ? 'red' : 'green')](+failed + ' Failed'));
 
       return writeReport(config, report).then(function (results) {
         for (var i = 0; i < results.length; i++) {
-          if (results[i].state != 'fulfilled') {
-            logger.error("Failed writing report with error: " + results[i].value);
+          if (results[i].state !== 'fulfilled') {
+            logger.error('Failed writing report with error: ' + results[i].value);
           }
         }
 
@@ -113,8 +113,8 @@ module.exports = {
           throw new Error('Mismatch errors found.');
         }
       });
-    }, function(e) {
-      logger.error("Comparison failed with error:" + e);
+    }, function (e) {
+      logger.error('Comparison failed with error:' + e);
     });
   }
 };

--- a/core/util/compare.js
+++ b/core/util/compare.js
@@ -56,7 +56,7 @@ module.exports = function (config) {
 
   var report = new Reporter(config.ciReport.testSuiteName);
 
-  return tests = map(compareConfig.testPairs, function (pair) {
+  return map(compareConfig.testPairs, function (pair) {
     var Test = report.addTest(pair);
 
     var referencePath = path.join(config.projectPath, pair.reference);
@@ -94,7 +94,7 @@ module.exports = function (config) {
       });
   }, { concurrency: config.asyncCompareLimit || ASYNC_COMPARE_LIMIT }).then(function () {
     return report;
-  }, function(e) {
-    logger.error("The comparison failed with error: " + e);
+  }, function (e) {
+    logger.error('The comparison failed with error: ' + e);
   });
 };

--- a/core/util/createBitmaps.js
+++ b/core/util/createBitmaps.js
@@ -20,13 +20,13 @@ function regexTest (string, search) {
  * @return {Promise}        Resolves when fs.writeFile has completed.
  */
 function writeReferenceCreateConfig (config, isReference) {
-    var configJSON;
+  var configJSON;
 
-    if( typeof config.args.config === "object" ) {
-        configJSON = config.args.config;
-    } else {
-        configJSON = require(config.backstopConfigFileName);
-    }
+  if (typeof config.args.config === 'object') {
+    configJSON = config.args.config;
+  } else {
+    configJSON = require(config.backstopConfigFileName);
+  }
 
   configJSON.isReference = isReference;
   configJSON.paths.tempCompareConfigFileName = config.tempCompareConfigFileName;

--- a/core/util/findExecutable.js
+++ b/core/util/findExecutable.js
@@ -1,7 +1,6 @@
 var path = require('path');
 
 module.exports = function (module, bin) {
-
   try {
     if (module === 'phantomjs-prebuilt') {
       return require('phantomjs-prebuilt').path;
@@ -13,6 +12,6 @@ module.exports = function (module, bin) {
     var pathToExecutableModule = pathToExecutableModulePackageJson.replace('package.json', '');
     return path.join(pathToExecutableModule, relativePathToExecutableBinary);
   } catch (e) {
-    throw new Error('Couldn\'t find executable for module "' + module + '" and bin "' + bin + '"\n' + e.message)
+    throw new Error('Couldn\'t find executable for module "' + module + '" and bin "' + bin + '"\n' + e.message);
   }
 };

--- a/core/util/makeConfig.js
+++ b/core/util/makeConfig.js
@@ -1,7 +1,7 @@
 var path = require('path');
 var extendConfig = require('./extendConfig');
 
-function projectPath(config) {
+function projectPath (config) {
   // Legacy mode, if the cwd is the backstop module
   if (config.backstop === process.cwd()) {
     console.log('BackstopJS is running in legacy mode.');
@@ -10,10 +10,10 @@ function projectPath(config) {
   return process.cwd();
 }
 
-function loadProjectConfig(command, options, config) {
+function loadProjectConfig (command, options, config) {
   // TEST REPORT FILE NAME
   var customTestReportFileName = options && (options.testReportFileName || null);
-  if(customTestReportFileName) {
+  if (customTestReportFileName) {
     config.testReportFileName = options.testReportFileName || null;
   }
 
@@ -30,25 +30,25 @@ function loadProjectConfig(command, options, config) {
 
   var userConfig = {};
   var CMD_REQUIRES_CONFIG = command !== 'genConfig';
-  if(CMD_REQUIRES_CONFIG) {
-      if(options && typeof options.config === 'object') {
-          console.log('BackstopJS uses a passed object as config');
-          userConfig = options.config;
-      } else if (config.backstopConfigFileName) {
-          try {
-              console.log('BackstopJS loading config: ', config.backstopConfigFileName, '\n');
-              userConfig = require(config.backstopConfigFileName);
-          } catch (e) {
-              console.error('Error ' + e);
-              process.exit(1);
-          }
+  if (CMD_REQUIRES_CONFIG) {
+    if (options && typeof options.config === 'object') {
+      console.log('BackstopJS uses a passed object as config');
+      userConfig = options.config;
+    } else if (config.backstopConfigFileName) {
+      try {
+        console.log('BackstopJS loading config: ', config.backstopConfigFileName, '\n');
+        userConfig = require(config.backstopConfigFileName);
+      } catch (e) {
+        console.error('Error ' + e);
+        process.exit(1);
       }
+    }
   }
 
   return userConfig;
 }
 
-function makeConfig(command, options) {
+function makeConfig (command, options) {
   var config = {};
 
   config.args = options || {};

--- a/core/util/runCasper.js
+++ b/core/util/runCasper.js
@@ -1,6 +1,4 @@
-var path = require('path');
 var spawn = require('child_process').spawn;
-var isWin = require('./isWin');
 var findExecutable = require('./findExecutable');
 
 function getCasperArgs (config, tests) {
@@ -16,8 +14,8 @@ function getCasperArgs (config, tests) {
       args = args.concat(config.casperFlags);
     }
   }
-  
-  if(config.args.user && config.args.password){
+
+  if (config.args.user && config.args.password) {
     args = args.concat(['--user=' + config.args.user, '--password=' + config.args.password]);
   }
 

